scripts/publish.sh

```diff
@@ -21,13 +21,33 @@ IMAGE_SHA_MATCHED="FALSE"
 AWS_FOR_FLUENT_BIT_VERSION=$(cat ../AWS_FOR_FLUENT_BIT_VERSION)
 AWS_FOR_FLUENT_BIT_STABLE_VERSION=$(cat ../AWS_FOR_FLUENT_BIT_STABLE_VERSION)
 
+PUBLISH_LATEST=$(cat ../linux.version | jq -r '.linux.latest')
+echo "Publish Latest? ${PUBLISH_LATEST}"
+
+# Problem: when we push a new version bump the version number in AWS_FOR_FLUENT_BIT_VERSION file changes
+# but that version is not published immediately. Thus, sync tasks normally
+# sync latest version found in DockerHub. But what if we want to release a non-latest version?
+# then sync tasks need to know this. So we use the script to check for an already published version
+# that's not latest
 docker_hub_image_tags=$(curl -s -S 'https://registry.hub.docker.com/v2/repositories/amazon/aws-for-fluent-bit/tags/?page=1&page_size=250' | jq -r '.results[].name')
 tag_array=(`echo ${docker_hub_image_tags}`)
-AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB=$(./get_latest_dockerhub_version.py linux ${tag_array[@]})
+AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB=$(./get_latest_dockerhub_version.py linux latest ${tag_array[@]})
+
+# If the AWS_FOR_FLUENT_BIT_VERSION is an older version which is already published to dockerhub
+# and latest is set to false in linux.version, then we sync an older non-latest version.
+# otherwise, normal behavior, sync latest version found in dockerhub
+if [ "${PUBLISH_LATEST}" = "false" ]; then
+	PUBLISH_NON_LATEST=$(./get_latest_dockerhub_version.py linux ${AWS_FOR_FLUENT_BIT_VERSION} ${tag_array[@]})
+	if [ "${PUBLISH_NON_LATEST}" = "true" ]; then
+		AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB=${AWS_FOR_FLUENT_BIT_VERSION}
+	fi
+fi
+
```

Comment:
@matthewfala I could add a check here to fail fast if the `AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB` is empty string. How could that case happen now though? I think it still shouldn't happen,

Reply:
I actually don't see any way that AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB can be an empty string any more. It will be latest if not the AWS_FOR_FLUENT_BIT_VERSION as you describe in your comment below.